diesel_codegen/src/schema_inference/pg.rs

```diff
@@ -74,17 +83,28 @@ pub fn load_table_names(
 pub fn get_table_data(conn: &PgConnection, table_name: &str) -> QueryResult<Vec<ColumnInformation>> {
     use self::pg_attribute::dsl::*;
     use self::pg_type::dsl::{pg_type, typname};
-    let t_oid = try!(table_oid(conn, table_name));
+    use diesel::expression::sql_literal::sql;
 
     pg_attribute.inner_join(pg_type)
         .select((attname, typname, attnotnull))
-        .filter(attrelid.eq(t_oid))
+        .filter(sql(&format!("pg_attribute.attrelid = '{}'::regclass",table_name)))
         .filter(attnum.gt(0).and(attisdropped.ne(true)))
         .order(attnum)
         .load(conn)
 }
 
-fn table_oid(conn: &PgConnection, table_name: &str) -> QueryResult<u32> {
-    use self::pg_class::dsl::*;
-    pg_class.select(oid).filter(relname.eq(table_name)).first(conn)
+pub fn get_primary_keys(conn: &PgConnection, table_name: &str) -> Vec<String> {
+    use self::pg_attribute::dsl::*;
+    use self::pg_index::dsl::{pg_index, indisprimary, indexrelid};
+    use diesel::expression::sql_literal::sql;
+
+    let pk_query = pg_index.select(indexrelid)
+        .filter(sql(&format!("pg_index.indrelid = '{}'::regclass",table_name)))
```

Comment:
Why the cast instead of the `table_oid` function that was there before? (I'd prefer to avoid literal SQL as it's unchecked by the compiler)